### PR TITLE
Support create replica_query rule.

### LINKED ISSUE #8615: Support CREATE REPLICA_QUERY RULE
```sql
CREATE REPLICA_QUERY RULE (
logic_ds_a (primary=primary_ds_a, repilca=repilca_ds_a_0, repilca_ds_a_1) RANDOM(xxx=xxx),
logic_ds_b (primary=primary_ds_b, repilca=repilca_ds_b_0, repilca_ds_b_1 ROUNDROBIN(xxx=xxx),
)
```

### PR DESCRIPTION
**Pull request by jingshanglu**
Fixes #8615 .

### CODE REVIEW

File: shardingsphere-distsql-parser/shardingsphere-distsql-parser-engine/src/main/antlr4/imports/RDLStatement.g4 [hunk 1]
```diff
@@ -63,6 +63,14 @@ createShardingRule
     : CREATE SHARDING RULE LP shardingTableRuleDefinition (COMMA shardingTableRuleDefinition)* RP
     ;
 
+createReplicaQueryRule
+    : CREATE REPLICA_QUERY RULE LP replicaQueryRule (COMMA replicaQueryRule)* RP
```
Comment (terrymanu) [inline shardingsphere-distsql-parser/shardingsphere-distsql-parser-engine/src/main/antlr4/imports/RDLStatement.g4:67]: replicaQueryRule should rename to replicaQueryRuleDefinition

File: shardingsphere-distsql-parser/shardingsphere-distsql-parser-engine/src/main/antlr4/imports/RDLStatement.g4 [hunk 2]
```diff
@@ -63,6 +63,14 @@ createShardingRule
...
+    ;
+
+replicaQueryRule
+    : rule_name=IDENTIFIER LP PRIMARY EQ primary_ds=schemaName COMMA REPLICA EQ schemaNames RP load_balancer=IDENTIFIER LP  funcProperties RP
```
Comment (terrymanu) [inline shardingsphere-distsql-parser/shardingsphere-distsql-parser-engine/src/main/antlr4/imports/RDLStatement.g4:71]: Please do not use underline to name variable

Comment (terrymanu) [inline shardingsphere-distsql-parser/shardingsphere-distsql-parser-engine/src/main/antlr4/imports/RDLStatement.g4:71]: primary_ds should rename to primary

File: shardingsphere-distsql-parser/shardingsphere-distsql-parser-engine/src/main/antlr4/imports/RDLStatement.g4 [hunk 3]
```diff
@@ -102,3 +110,23 @@ columName
...
+
+schemaNames
+    : schemaName (COMMA schemaName)*
+    ;
+
+schemaName
+    : IDENTIFIER
+    ;
+
+funcProperties
```
Comment (terrymanu) [inline shardingsphere-distsql-parser/shardingsphere-distsql-parser-engine/src/main/antlr4/imports/RDLStatement.g4:126]: What's mean of funcProperties, where is the `func`?

File: shardingsphere-distsql-parser/shardingsphere-distsql-parser-engine/src/main/antlr4/imports/RDLStatement.g4 [hunk 4]
```diff
@@ -102,3 +110,23 @@ columName
...
+    : funcPropertie (COMMA funcPropertie)*
+    ;
+
+funcPropertie
```
Comment (terrymanu) [inline shardingsphere-distsql-parser/shardingsphere-distsql-parser-engine/src/main/antlr4/imports/RDLStatement.g4:130]: `Propertie` is typo

File: shardingsphere-distsql-parser/shardingsphere-distsql-parser-engine/src/main/antlr4/org/apache/shardingsphere/distsql/parser/autogen/DistSQLStatement.g4 [hunk 5]
```diff
@@ -25,5 +25,6 @@ execute
     | dropShardingRule
     | showResources
     | showRule
+    | createReplicaQueryRule
```
Comment (terrymanu) [inline shardingsphere-distsql-parser/shardingsphere-distsql-parser-engine/src/main/antlr4/org/apache/shardingsphere/distsql/parser/autogen/DistSQLStatement.g4:28]: Please move it about showXXX

**Review by terrymanu — CHANGES_REQUESTED** [review]

File: shardingsphere-distsql-parser/shardingsphere-distsql-parser-engine/src/main/antlr4/imports/RDLStatement.g4 [hunk 6]
```diff
@@ -63,42 +63,42 @@ createShardingRule
...
-shardingTableRuleDefinition
-    : tableName columName shardingAlgorithmDefinition
+createReplicaQueryRule
+    : CREATE REPLICA_QUERY RULE LP replicaQueryRuleDefinition (COMMA replicaQueryRuleDefinition)* RP
     ;
 
-shardingAlgorithmDefinition
-    : shardingAlgorithmType LP shardingAlgorithmProperties RP
+replicaQueryRuleDefinition
+    : ruleName=IDENTIFIER LP PRIMARY EQ primary=schemaName COMMA REPLICA EQ schemaNames RP loadBalancer=IDENTIFIER LP  algorithmProperties RP
```
Comment (terrymanu) [inline shardingsphere-distsql-parser/shardingsphere-distsql-parser-engine/src/main/antlr4/imports/RDLStatement.g4:71]: There are 2 spaces between `LP  algorithmProperties`

File: shardingsphere-distsql-parser/shardingsphere-distsql-parser-statement/src/main/java/org/apache/shardingsphere/distsql/parser/segment/rdl/ReplicaQueryRuleSegment.java [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+import lombok.Setter;
+import org.apache.shardingsphere.sql.parser.api.visitor.ASTNode;
+
+import java.util.Collection;
+import java.util.Properties;
+
+/**
+ * Table rule segment.
+ */
+@RequiredArgsConstructor
```
Comment (terrymanu) [inline shardingsphere-distsql-parser/shardingsphere-distsql-parser-statement/src/main/java/org/apache/shardingsphere/distsql/parser/segment/rdl/ReplicaQueryRuleSegment.java:31]: What is useful of `@RequiredArgsConstructor`?

File: shardingsphere-distsql-parser/shardingsphere-distsql-parser-statement/src/main/java/org/apache/shardingsphere/distsql/parser/statement/rdl/create/impl/CreateReplicaQueryRuleStatement.java [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import org.apache.shardingsphere.distsql.parser.segment.rdl.ReplicaQueryRuleSegment;
+import org.apache.shardingsphere.distsql.parser.statement.rdl.create.CreateRDLStatement;
+
+import java.util.Collection;
+
+/**
+ * Create sharding rule statement.
```
Comment (terrymanu) [inline shardingsphere-distsql-parser/shardingsphere-distsql-parser-statement/src/main/java/org/apache/shardingsphere/distsql/parser/statement/rdl/create/impl/CreateReplicaQueryRuleStatement.java:28]: The javadoc is incorrect

File: shardingsphere-features/shardingsphere-replica-query/shardingsphere-replica-query-common/pom.xml [hunk 9]
```diff
@@ -29,6 +29,12 @@
     <name>${project.artifactId}</name>
     
     <dependencies>
+        <dependency>
+            <groupId>org.apache.shardingsphere</groupId>
+            <artifactId>shardingsphere-infra-binder</artifactId>
+            <version>${project.version}</version>
+        </dependency>
+
```
Comment (terrymanu) [inline shardingsphere-features/shardingsphere-replica-query/shardingsphere-replica-query-common/pom.xml:37]: What is the useful?

File: shardingsphere-features/shardingsphere-replica-query/shardingsphere-replica-query-common/src/main/java/org/apache/shardingsphere/replicaquery/yaml/converter/CreateReplicaQueryRuleStatementConverter.java [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+import org.apache.shardingsphere.replicaquery.yaml.config.rule.YamlReplicaQueryDataSourceRuleConfiguration;
+
+/**
+ * Create replica query rule statement converter.
+ */
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+public final class CreateReplicaQueryRuleStatementConverter {
+    
+    /**
+     * Convert create replica query rule statement context to YAML sharding rule configuration.
```
Comment (terrymanu) [inline shardingsphere-features/shardingsphere-replica-query/shardingsphere-replica-query-common/src/main/java/org/apache/shardingsphere/replicaquery/yaml/converter/CreateReplicaQueryRuleStatementConverter.java:35]: context to YAML `sharding` rule configuration?

**Review by terrymanu — CHANGES_REQUESTED** [review]

File: shardingsphere-features/shardingsphere-replica-query/shardingsphere-replica-query-common/pom.xml (hunk 9 above)
Comment (jingshanglu) [reply shardingsphere-features/shardingsphere-replica-query/shardingsphere-replica-query-common/pom.xml]: @terrymanu `Createreplicaqueryrulestatementconverter ` use `createreplicaqueryrulestatement ` I have change it to `shardingsphere-distsql-parser-statement`.

File: shardingsphere-distsql-parser/shardingsphere-distsql-parser-engine/src/main/java/org/apache/shardingsphere/distsql/parser/core/DistSQLVisitor.java [hunk 11]
```diff
@@ -86,29 +91,57 @@ public ASTNode visitCreateShardingRule(final CreateShardingRuleContext ctx) {
...
         PropertiesValue result = new PropertiesValue();
-        for (ShardingAlgorithmPropertyContext each : ctx.shardingAlgorithmProperty()) {
-            result.getValue().setProperty(each.shardingAlgorithmPropertyKey().getText(), each.shardingAlgorithmPropertyValue().getText());
+        for (AlgorithmPropertyContext each : ctx.algorithmProperty()) {
+            result.getValue().setProperty(each.key.getText(), each.value.getText());
         }
         return result;
     }
-    
+
```
Comment (terrymanu) [inline shardingsphere-distsql-parser/shardingsphere-distsql-parser-engine/src/main/java/org/apache/shardingsphere/distsql/parser/core/DistSQLVisitor.java:144]: Please keep the original indent

**Review by terrymanu — CHANGES_REQUESTED** [review]

**Review by terrymanu — APPROVED** [review]

**Pull request merged**